begin draft algorithm paper (#18)
nptscot/rnetmatch: paper.qmd

---

> title: "Approximate Network Matching"

Reviewer: Great to get this written up!

author: 
  - name: Josiah Parry
    affiliation: Environmental Systems Research Institute, Redlands, CA, USA 
    orcid: 0000-0001-9910-865X
  - name: Robin Lovelace
    affiliation: Leeds Institute for Transport Studies, University of Leeds, UK
    orcid: 0000-0001-5679-6536
format: html
filters:
  - pseudocode
---

## Abstract 

> - A common issue in spatial data science is the reconciliation of two sets of linestrings. 

Reviewer: Maybe start broader:

```suggestion
- A common need in spatial data science is spatial joins.
```

- linestrings may represent the same phenomenon but be topologically different
- joining data between these road networks is problematic for many reasons
- often there may be no join key present
  - if there is a join key, there remains an issue of wholesale joining of attributes
    - attributes of a linestring are intended be associated with that linestring, not another
    - we need to do a join and provide a weight for future calculations 
- in this paper we introduce a new algorithm to match two sets of linestrings
- it is implemented in an open source rust library with bindings to R and Python

## Problem Statement 

TODO 

## Existing algorithms 

## Algorithm overview 

> The proposed algorithm aims to match elements of two sets of LineStrings that are topologically similar and estimate the amount of shared length between each pair of matched line strings.

Reviewer: :+1:
